extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -411,7 +411,14 @@ void adaptOpenTelemetryJdbcInstrumentationForNative(BuildProducer<RemovedResourc
     }
 
     @BuildStep
-    NativeImageResourceBundleBuildItem registerRowSetResourceBundle() {
-        return new NativeImageResourceBundleBuildItem("com.sun.rowset.RowSetResourceBundle");
+    void registerRowSetSupport(
+            BuildProducer<NativeImageResourceBundleBuildItem> resourceBundleProducer,
+            BuildProducer<NativeImageResourceBuildItem> nativeResourceProducer,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassProducer) {
+        resourceBundleProducer.produce(new NativeImageResourceBundleBuildItem("com.sun.rowset.RowSetResourceBundle"));
+        nativeResourceProducer.produce(new NativeImageResourceBuildItem("javax/sql/rowset/rowset.properties"));
+        reflectiveClassProducer.produce(ReflectiveClassBuildItem.builder(
+                "com.sun.rowset.providers.RIOptimisticProvider",
+                "com.sun.rowset.providers.RIXMLProvider").build());
```

Comment:
This I'm not totally sure as it might bring a lot of XML classes with it.

This needs to be checked by compiling a small app with Agroal (and a datasource) and check the size of the native executable before and after the commit.

Reply:
Added the output here: https://github.com/quarkusio/quarkus/pull/38370#issuecomment-1908307782